src/test/scala/com/cognite/sdk/scala/v1/fdm/instances/CogniteAssetsTest.scala

```diff
@@ -0,0 +1,102 @@
+package com.cognite.sdk.scala.v1.fdm.instances
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import com.cognite.sdk.scala.v1.fdm.Utils
+import com.cognite.sdk.scala.v1.fdm.instances.InstanceDeletionRequest.NodeDeletionRequest
+import com.cognite.sdk.scala.v1.fdm.instances.NodeOrEdgeCreate.NodeWrite
+import com.cognite.sdk.scala.v1.fdm.views.ViewReference
+import com.cognite.sdk.scala.v1.{CommonDataModelTestHelper, File, FileDownloadInstanceId, FileDownloadLink, FileUploadInstanceId, InstanceId}
+import sttp.client3.UriContext
+
+import java.io.{BufferedInputStream, FileInputStream}
+import scala.annotation.tailrec
+import scala.util.Try
+
+@SuppressWarnings(
+  Array(
+    "org.wartremover.warts.NonUnitStatements",
+    "org.wartremover.warts.ThreadSleep",
+    "org.wartremover.warts.Equals"
+  )
+)
+class CogniteAssetsTest extends CommonDataModelTestHelper {
+  private val defaultAttemptCount: Int = 7
+
+  it should "make it possible to retrieve file and associated upload link and download link using instance id" in {
+
+    val instanceId: InstanceId = InstanceId(space = Utils.SpaceExternalId, externalId = "file_instance_ext_id2")
+
+    val testFile = InstanceCreate(
+      items = Seq(
+        NodeWrite(
+          space = instanceId.space,
+          externalId = instanceId.externalId,
+          Some(Seq(EdgeOrNodeData(ViewReference("cdf_cdm", "CogniteFile", "v1"), None))
+          ),
+          None
+        )),
+      None,
+      None,
+      None,
+      None
+    )
+    val createdItem = testClient.instances.createItems(testFile).unsafeRunSync()
+    val retrievedItem =
+      retry[Seq[File]](() => testClient.files.retrieveByInstanceIds(Seq(instanceId)), files => files.headOption.map(_.createdTime).nonEmpty, defaultAttemptCount)
+
+    val retrievedSingleItem =
+      retry[File](() => testClient.files.retrieveByInstanceId(instanceId), _ => true, defaultAttemptCount)
+
+    val uploadLinkFile =
+      retry[File](() => testClient.files.uploadLink(FileUploadInstanceId(instanceId)), file => file.uploadUrl.nonEmpty, defaultAttemptCount)
+
+    uploadLinkFile shouldBe a [Right[_, _]]
+
+    val file = new java.io.File("./src/test/scala/com/cognite/sdk/scala/v1/uploadTest.txt")
+    val inputStream = new BufferedInputStream(
+      new FileInputStream(
+        file
+      )
+    )
+    val fileSize = file.length()
+    uploadLinkFile.map(_.uploadUrl) match {
+      case Right(Some(uploadUrl)) =>
+        testClient.requestSession.send { request =>
+          request
+            .contentLength(fileSize)
```

Comment:
when setting the contentLength, I guess the backend stops trying to chunk the content, which means there won't be a transfer-encoding set, which means this actually works in azure too.

Not sure if there is a better way to enforce that, but nothing I've tried works.

For example, trying to do .header("Transfer-Encoding", None) doesn't work, using disableAutoCompression also doesn't work.